lib/jenkins_pipeline_builder/pull_request_generator.rb

```diff
@@ -19,149 +19,53 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 #
-
 module JenkinsPipelineBuilder
   class PullRequestGenerator
     class NotFound < StandardError; end
+    attr_accessor :open_prs, :application_name
 
-    attr_reader :purge, :create, :jobs, :project, :generator, :pull_generator, :errors, :pull_requests
-
-    def initialize(project, generator)
-      @project = project
-      @generator = generator
-      @purge = []
-      @create = []
-      @errors = {}
-
-      @pull_generator = find
-      payload = compile_generator
-      return unless payload
-      @jobs = filter_jobs
-
-      # old init
-      @pull_requests = check_for_pull payload
-      find_old_pull_requests
-      generate_pull_requests
-
-      collection.merge! @jobs
-      @errors.merge! create_jobs
-
-      purge_jobs
-    end
-
-    def valid?
-      errors.empty?
-    end
-
-    private
-
-    def generate_pull_requests
-      @pull_requests.each do |number|
-        req = JenkinsPipelineBuilder::PullRequest.new(project, number, jobs, @pull_generator)
-        @jobs.merge! req.jobs
-        project_new = req.project
-
-        # Overwrite the jobs from the generator to the project
-        project_new[:value][:jobs] = req.jobs.keys
-        @create << project_new
-      end
+    def initialize(application_name: nil, github_site: nil, git_org: nil, git_repo_name: nil)
+      @application_name = application_name
+      @open_prs = active_prs git_url: github_site, git_org: git_org, git_repo: git_repo_name
```

Comment:
I'd prefer that we store the defaults as `@options` or something instead of passing a bunch of parameters around. Alternately, we can store the parameters individually as instance vars.